pool/src/pool.rs

```diff
@@ -159,6 +163,50 @@ impl Pool {
 		Ok(Some(tx))
 	}
 
+	/// evict a transaction from the pool
+	pub fn evict_from_pool(&mut self) {
+		// Check that the new tx is not a bullshit tx in term of fee
+
+		let mut ordered_entries = self.entries.clone();
+		ordered_entries.sort_by(|entry_a, entry_b| {
+			let fw_a = entry_a.tx.fee() * 1000 / entry_a.tx.tx_weight() as u64;
+			let fw_b = entry_b.tx.fee() * 1000 / entry_b.tx.tx_weight() as u64;
+			if fw_a > fw_b {
+				Ordering::Greater
+			} else if fw_a == fw_b && entry_a.tx_at < entry_b.tx_at {
+				Ordering::Greater
+			} else {
+				Ordering::Less
+			}
+		});
+
+		// remove first one iff not a bucket root of several txs
```

Comment:
This is O(n^2) in number of transactions in the pool. Definitely needs to be optimized.

Reply:
Agree. Looking into this.